### ISSUE TITLE
Heroku app crashes after calls 

### ISSUE THREAD

**Issue by johnny-stevie** [HermanFassett/youtube-scrape#12]
@HermanFassett Thank you for this library! It has made my life very easy! I am using it to create an app! However, the problem is that the heroku api crashes after more than 20-ish requests. I wanted to make sure that this was heroku's fault due to request overload. Do you think this lib is robust enough to take many requests at once once my app is in production mode? It would be a shame if i had to change youtube-search provider.

**Comment by HermanFassett** [comment]
Hey, my heroku deployment does have limited usage because it's on a free tier so that could be part of the issue. Having your own server hosting this would make it more reliable.
However, I've heard this approach could have an issue with YouTube itself limiting the web requests. I haven't verified it as I never reached a limit on local development, but I think someone said in another issue they got cut off at around 150 requests.
If that's so, the only way you'll ever get a prod ready solution would be using the official YouTube API. It's disappointing as I didn't realize this in development. Again, I haven't verified this is an issue, but it sounds likely. Perhaps there's a way around it, as I've never heard of a regular browser user getting limited.
Until more is known it would be safe to assume that this is a handy project for someone who wants to prototype, but I would look into using the official API for production apps.

**Comment by johnny-stevie** [comment]
Doesnt the official Youtube api has also its own quota limitations? That would be annoying for people wanting to use it in production. However, I did use web scraping to get the youtube search results but it is a slow process due to the large html content you get from a youtube search. I have tried using NewPipeExtractor as its a good option for scraping youtube search info and downloading quickly but its hard to implement in my app. Thanks anyway!
